lib/ansiblelint/text.py

```diff
@@ -0,0 +1,76 @@
+"""Text utils."""
+import re
+import subprocess
+from typing import TYPE_CHECKING, Any, List, Union
+
+from ansiblelint._internal.rules import BaseRule, RuntimeErrorRule
+from ansiblelint.errors import MatchError
+from ansiblelint.rules.AnsibleSyntaxCheckRule import AnsibleSyntaxCheckRule
+
+if TYPE_CHECKING:
+    # https://github.com/PyCQA/pylint/issues/3240
+    # pylint: disable=unsubscriptable-object
+    CompletedProcess = subprocess.CompletedProcess[Any]
+else:
+    CompletedProcess = subprocess.CompletedProcess
+
+_ansible_syntax_check_re = re.compile(
+    r"^ERROR! (?P<title>[^\n]*)\n\nThe error appears to be in "
+    r"'(?P<filename>.*)': line (?P<line>\d+), column (?P<column>\d+)",
+    re.MULTILINE | re.S | re.DOTALL)
+
+
+def strip_ansi_escape(data: Union[str, bytes]) -> str:
+    """Remove all ANSI escapes from string or bytes.
+
+    If bytes is passed instead of string, it will be converted to string
+    using UTF-8.
+    """
+    if isinstance(data, bytes):
+        data = data.decode("utf-8")
+
+    return re.sub(r"\x1b[^m]*m", "", data)
+
+
+def parse_ansible_syntax_check(run: CompletedProcess) -> List[MatchError]:
+    """Convert ansible syntax check output into a list of MatchError(s)."""
+    result = []
+    if run.returncode != 0:
```

Comment:
This is useless since this function should not be called when there is nothing wrong with the input. Replacing the `run` input parameter with some pure text would solve this issue as a side-effect.

Reply:
Is not really useless as the result of running syntax check is not a single string value, is a tuple formed of exit code, stderr and stdout content. Differences between the stream are likely to be relevant in the future, when the warnings will finally become real warnings, maybe even machine parsable if we are lucky.

The exit code is also relevant because currently value 4 means syntax check failed, 0 success and other values indicate other Ansible errors.